### ISSUE TITLE
[Bug]: Memory Leak Issue

### ISSUE THREAD

**Issue by KristofferEriksson** [rnmapbox/maps#2591]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### Platform

iOS, Android

### `@rnmapbox/maps` version

#main

### Standalone component to reproduce

```javascript
import React from 'react';
import {
  MapView,
  ShapeSource,
  LineLayer,
  Camera,
} from '@rnmapbox/maps';

const aLine = {
  type: 'LineString',
  coordinates: [
    [-74.00597, 40.71427],
    [-74.00697, 40.71527],
  ],
};

class BugReportExample extends React.Component {
  render() {
    return (
      <MapView style={{flex: 1}}>
        <Camera centerCoordinate={[-74.00597, 40.71427]} zoomLevel={14} />
        <ShapeSource id="idStreetLayer" shape={aLine}>
          <LineLayer id="idStreetLayer" />
        </ShapeSource>
      </MapView>
    );
  }
}
```

### Observed behavior and steps to reproduce

When navigating to a screen with a map and then navigating back or to another screen, the map does not unload properly. Every time we're loading a screen with a map in it, the memory adds up until it causes a memory exhaustion crash. This occurs both on iOS and on Android.

<img width="1115" alt="Skærmbillede 2023-02-07 kl  14 40 33" src="https://user-images.githubusercontent.com/44787769/217260841-68edca1d-c88b-4ef6-8522-2232e37ff810.png">

**Comment by thibaultitautomotive** [comment]
Same issue on my project

**Comment by bradstiff** [comment]
I also have this issue.

**Comment by mfazekas** [comment]
@KristofferEriksson I've tried on current master, one with 10.11 as mapbox dependency. And It was working me just fine. I've replace BugReportExample.js with the code you've pasted bellow and then opened closed the BugReportTemplate example.

![image](https://user-images.githubusercontent.com/52435/219044081-ed065791-459f-47df-a970-3c3f70a730f0.png)

**Comment by mfazekas** [comment]
I'm closing this as cannot reproduce.

**Comment by mfazekas** [comment]
@bradstiff added a fix for that - #2707

**Comment by bradstiff** [comment]
Beautiful. Looking forward to the release!
